extensions/windows-event-log/wel/MetadataWalker.cpp

```diff
@@ -17,21 +19,62 @@ bool MetadataWalker::for_each(pugi::xml_node &node) {
 
 			if (std::regex_match(attr.name(), regex_)) {
 				std::function<std::string(const std::string &)> idUpdate = [&](const std::string &input) -> std::string {
-					return resolve_ ? utils::OsUtils::userIdToUsername(input) : input;
+					if (resolve_) {
+						auto resolved = utils::OsUtils::userIdToUsername(input);
+						replaced_identifiers_[input] = resolved;
+						return resolved;
+					}
+					else {
+						replaced_identifiers_[input] = input;
```

Comment:
This else branch has no return value. Which surely triggers a warning, but I'm afraid it's an undefined behaviour.

Reply:
@am-c-p-p can you take a look in this to correct?